Take into account citation when transform a quote to paragraph

oandregal opened the pull request:
When transforming a quote to a paragraph the citation was left out. This PR makes the citation a new paragraph.

oandregal committed:
Take into account citation when transform a quote to paragraph

iseulde commented on packages/block-library/src/quote/index.js:
```diff
@@ -97,8 +97,8 @@ export const settings = {
 				blocks: [ 'core/paragraph' ],
-				transform: ( { value } ) =>
-					split( create( { html: value, multilineTag: 'p' } ), '\u2028' )
+				transform: ( { value, citation } ) =>
```
This can be added after split, no? No need to parse it.

iseulde commented:
Could you added an e2e test as well?

oandregal committed:
Refactor to take into account blank quote or cite

- quote + cite should render at least two paragraphs (more if quotes contains several paragraphs)
- void quote + cite should render only one paragraph with the cite
- quote + void cite should render only one paragraph with the quote

oandregal committed:
Protect against void quote and cite

oandregal committed:
Update e2e tests

nosolosw commented:
@iseulde pushed some changes to make it more robust against void cite, quotes, or both. Also updated the existing e2e tests.

ellatrix reviewed (approved):
Ok, looks good.

iseulde commented:
@nosolosw Oh, could we do the same for pull quote?

nosolosw commented:
Cool! Going to merge.

Pullquote can only be transformed to/from a quote since [a few hours ago](https://github.com/WordPress/gutenberg/pull/10683) :tada: Depending on how clean is [the queue](https://github.com/WordPress/gutenberg/milestone/66) tomorrow morning I might be able to add the pullquote<->paragraph transform (and take the cite into account).

paaljoachim commented:
https://make.wordpress.org/test/
13. Add a quote block with a citation, convert it to a paragraph block, make sure the citation is still there. (10685)

Works. Should the citation come direct below the quote in the same paragraph box? Or should it as it is right now go into another paragraph box below the quote?

nosolosw commented:
> Should the citation come direct below the quote in the same paragraph box? Or should it as it is right now go into another paragraph box below the quote?

The expected behavior is that a new paragraph is created for the cite.

paaljoachim commented:
Ok.

mikelittle commented:
That worked for me too.
